pkg/pipeline/source/sdk.go

```diff
@@ -639,8 +639,9 @@ func (s *SDKSource) onTrackFinished(trackID string) {
 		writer.Drain(true)
 		active := s.active.Dec()
 		if s.RequestType == types.RequestTypeParticipant || s.RequestType == types.RequestTypeRoomComposite {
-			s.callbacks.OnTrackRemoved(trackID)
 			s.sync.RemoveTrack(trackID)
+			<-s.callbacks.BuildReady
```

Comment:
Same here, any other channel we can select on?

Reply:
not for this one, this needs to happen even if `closed` was already broken